src/runtime_src/core/common/drv/kds_core.c

```diff
@@ -477,6 +495,42 @@ kds_submit_ert(struct kds_sched *kds, struct kds_command *xcmd)
 	return 0;
 }
 
+static int
+kds_test_and_refcnt_incr(struct kds_client *client, int domain, int cu_idx)
+{
+	int prev;
+	u32 *refs;
+
+	if (domain == DOMAIN_PL || domain == DOMAIN_VIRT)
```

Comment:
do we have a specific domain for virtual cu? if we do, we map cu_refs to it. But in line 271, we map scu_refs to it? Maybe we should have a function to map cu_/scu_refs to domain and be consistent with our design?